lib/AST/ASTImporter.cpp

```diff
@@ -2201,28 +2201,29 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
         if (!ToDescribed)
           return nullptr;
         D2CXX->setDescribedClassTemplate(ToDescribed);
-        if (!DCXX->isInjectedClassName() && D->isCompleteDefinition()) {
+        if (!DCXX->isInjectedClassName()) {
           // In a record describing a template the type should be a
           // InjectedClassNameType (see Sema::CheckClassTemplate). Update the
           // previously set type to the correct value here (ToDescribed is not
           // available at record create).
           // FIXME: The previous type is cleared but not removed from
           // ASTContext's internal storage.
-          D2CXX->setTypeForDecl(nullptr);
-          Importer.getToContext().getInjectedClassNameType(D2CXX,
-              ToDescribed->getInjectedClassNameSpecialization());
-          // Find the injected class name itself and update the type too.
-          CXXRecordDecl *Injected = [&] {
+          auto Injected = [&]()->CXXRecordDecl * {
             for (NamedDecl *Found : D2CXX->noload_lookup(Name)) {
               auto *Record = dyn_cast<CXXRecordDecl>(Found);
               if (Record && Record->isInjectedClassName()) {
                 return Record;
               }
             }
-            assert(false && "No injected-class-name found!");
+            return nullptr;
```

Comment:
When will this return a nullptr? In case this only reached for non-complete definitions, maybe we could assert that.

Reply:
In nullptr case the injected-class-name is not yet imported. But it should be imported later and then the already created InjectedClassNameType is used.